contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault, IDepositContract} from "../interfaces/IRibbon.sol";
+import {ICRV} from "../interfaces/ICRV.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+contract VaultQueue is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    enum TransferAction {INTERVAULT, WITHDRAW}
+    struct Transfer {
```

Comment:
To optimize this more, we can fit the Struct into 3 32-byte slots with [struct packing](https://fravoll.github.io/solidity-patterns/tight_variable_packing.html) 

```solidity
struct Transfer {
  // slot 1
  address creditor;
  // slot 2
   address dstVault;
  TransferAction action;
  uint32 timestamp;
  // slot 3
  uint256 amount;
}
```

Reply:
done